### PR TITLE
ci: update pr template to match rook's pr template

### PR DESCRIPTION
**Pull request by subhamkrai**
updating pr templatet to match the rook's pr template

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] CI tests has been updated, if necessary.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+**Issue resolved by this Pull Request:**
 Resolves #
+--->
 
 **Checklist:**
 
 - [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
 - [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
-- [ ] CI tests has been updated, if necessary.
+- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
```
Comment (Madhu-1) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Is this valid? Are we going to update Rook pending notes when we have a PR here?

Comment (subhamkrai) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: well, we do mention it in release notes but not sure we need it here, I just copied it from rook/rook. @travisn thoughts?

Comment (travisn) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: We don't need the checklist items to match from the rook repo where they don't apply. Let's remove the note about pending release notes.

Comment (subhamkrai) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**